File: src/main/scala/com/vc/medium/L495.scala
package com.vc.medium

object L495 {
  def findPoisonedDuration(timeSeries: Array[Int], duration: Int): Int = {
    var res = 0
    var newTime = 0
    var addDuration = 0
    timeSeries.foreach(time => {
      if(newTime > time) {
        addDuration = duration - (newTime - time)
        res += addDuration
      }
      else res += duration
      newTime = time + duration
    })
    res
  }
}

File: src/main/java/com/vc/medium/L495.java
package com.vc.medium;

class L495 {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int total = 0, prev = 0;
        for(int time: timeSeries) {
            if(prev >= time) {
                total += duration - (prev - time);
            }
            else total += duration;
            prev = time + duration;
        }
        return total;
    }
}
